godotproject/Menus/InGameMenu/InGameMenu.gd

```diff
@@ -0,0 +1,106 @@
+extends MarginContainer
+
+export (bool) var pause_child = false
+export (NodePath) var default_screen_path
+onready var default_screen = get_node(default_screen_path)
+
+func init_screen_list():
+	# This is a list of screens in the current menu.
+	screens.append($PauseSubmenu)
+	screens.append($HelpSubmenu)
+	screens.append($QuitVerifySubmenu)
+
+var screens = []
+var screen_current = null
+var screen_after_fade = null
+
+func update_active_screen():
+	# Pause and hide all of our child screens
+	for screen in screens:
+		screen.pause_mode = PAUSE_MODE_STOP
+		screen.visible = false
+	# If we're trying to display a null screen, we're trying to close the menu
+	if screen_current == null:
+		# Hide ourselves
+		visible = false
+		# Unpause the other game logic
+		get_tree().paused = false
+	else:
+		# Set pause on SceneTree - this will pause the game logic
+		# (except for us, because we have our pause mode set to Process, not Inherit)
+		get_tree().paused = true
+		# Show ourselves
```

Comment:
by "show ourselves" you mean show the menu screen right?

Reply:
Yep, that's correct
```suggestion
		# Show ourselves (the in-game menu)
```

Reply:
yup! jsut wanted to be sure I understood what was happening there